### PR TITLE
Fix failing doxygen GitHub action: deploy step

### PR DESCRIPTION
**Pull request by slfuchs**

### CODE REVIEW

**Comment by slfuchs** [comment]
@peterrum This should now be the solution as described by you in https://github.com/PartExa/PartExa/pull/12#issuecomment-994040115, correct?

File: .github/workflows/doxygen.yml [hunk 1]
```diff
@@ -39,6 +38,10 @@ jobs:
           if [ ! -f doxygen.err ]; then echo "No Doxygen error file found!"; exit 1; fi
           if [ -s doxygen.err ]; then echo "Doxygen failed with the following error(s):"; cat doxygen.err; exit 1; else echo "Doxygen passed."; fi
 
+  deploy:
```
Comment (peterrum) [inline .github/workflows/doxygen.yml:41]: Not completely. I guess you cannot access the created data of the `build` step here. Either GitHub allows you to do that somehow or you have generate the documentation again.

File: .github/workflows/doxygen.yml [hunk 2]
```diff
@@ -39,6 +38,10 @@ jobs:
...
+    runs-on: ubuntu-latest
+
+    steps:
```
Comment (peterrum) [inline .github/workflows/doxygen.yml:44]: I guess for sake of experimenting, you could disable `if: ${{ github.event_name == 'push' || github.event_name == 'schedule' }}`.

Comment (slfuchs) [reply .github/workflows/doxygen.yml:44]: I had this disabled before, but it seems that actions making use of such `${{ secrets.ACTIONS_DEPLOY_KEY_DOXYGEN }}` a key may only be run on `push` or `schedule`.

File: .github/workflows/doxygen.yml (hunk 1 above)
Comment (slfuchs) [reply .github/workflows/doxygen.yml:41]: Okay, do you mind if we try #13 first?

**Pull request closed without merging**